readme.txt
[center][img width=600 height=80]http://www.smfsimple.com/img/ultimateportallogo.png[/img][/center][center][color=orange][size=12pt][b]Ultimate Portal By Smfsimple Ya esta Aqui![/b][/size][/color][/center][center]Ultimate Portal 0.3 esta disponible para la version estable de SMF. [color=red][b]SMF 2.0[/b][/color][/center][color=orange][size=12pt][b]Caracteristicas generales UP 0.4:[/b][/size][/color]
o Es totalmente gratuito! 
o Consta de soporte gratuito las 24hs del dia!
o Se adapta perfectamente a cualquier estilo que tu foro tenga.
o La administracion es sumamente intuitiva, no necesitaras ser un experto para conseguir lo que quieras para tu foro!

[color=orange][size=12pt][b]Que tiene la version 0.4 que no tenga la 0.3?[/b][/size][/color]
La version 0.4 de ultimate Portal se redise&#241;o para que el portal tenga un mejor rendimiento y dise&#241;o en todo aspecto.
Se agrego el idioma espa&#241;ol por defecto para que no tengas que instalar paquetes de idiomas aparte.
Se repararon peque&#241;os bugs encontrados que afectaban al ssi y al wap.

[color=orange][size=10pt][b]Nuevos bloques por defecto de la version 0.3![/b][/size][/color]
o Bloque "Ultimos Usuarios" (Con este bloque tendras un listado de los ultimos usuarios registrados en tu foro)
o Bloque "Temas recientes" (Muestra tus temas mas recientes en un bloque de tu portal) 

[color=orange][size=12pt][b]Caracteristicas Avanzadas UP 0.3[/b][/size][/color]
o Podras administrar tus bloques y editarlos a tu medida con un sofisticado sistema de edicion unico!
o Podras desactivar o activar UP con un solo click.  
o Podras editar tus archivos de lenguaje directamente desde la administracion del porta!  
o Podras editar los permisos para cada rango de usuario. Tu puedes ocultar/mostrar modulos en funcion de lo que tu foro requiera. 
o Tendras la opcion de activar el PORTAL MENU, quien se encargara de generar un menu muy elegante tanto en un bloque como tambien en el top de tu foro, suplantando al slogan de tu foro. 
o Podras usar las mejores tecnicas SEO directamente configurando UP con unos pocos clicks! Configura tu robot.txt, tus keywords, introduce tu codigo Google Analytics y muchas opciones mas! 

[hr]
[center][color=orange][size=12pt][b]Y como si esto fuera poco! Todavia queda muchisimo mas para contarles![/b][/size][/color][/center][color=orange][size=12pt][b]Nuevas Caracteristicas UP 0.3[/b][/size][/color]
o MultiBlok Up! Con multiblok Up podras tener el portal que tanto querias, adaptando mas de 1 bloque por columna o fila, lo que hara que el estilo de tu portada sea unico y no redundante como hasta ahora los conociamos!
o Reduccion de sobrecarga de tu sitio mediante 1 solo click! De esta manera tu podras reducir la sobrecarga (las consultas a la base de datos es solo una ves cada 30 minutos) del Portal.
o Nuevo estilo del administrador de modulos![center][color=orange][size=12pt][b]Queres mas?[/b][/size][/color][/center]o Edicion de las cadenas de idioma desde el panel de administracion: olvidate de temblar al abrir archivos php.
o Creacion de paginas internas: Si quieres agregar informacion extra, hazlo desde la comodidad del Panel del Ultimate Portal.
o SEO: Seguimiento de estadisticas, improve title, Creacion de archivo robots.txt
o Permisos: Decide quien puede ver determinados bloques, con un par de clicks.
o Bloque afiliados, Seccion Quienes Somos (About us), Sistema de descargas, chat,..

Ademas es un completo portal con caracteristicas de linksharing, con el cual no tendras que hacer mayores cambios para compartir informacion.
o Novedades en portada, con un novedoso estilo[center][color=orange][size=12pt][b]Bloques por defecto de Ultimate Portal[/b][/size][/color][/center]o Afiliados
o Menu del Portal
o Noticias (Bloque propio del Modulo Noticias)
o Usuarios en Linea
o Menu Personal
o Estadisticas del Sitio

[center][color=orange][size=12pt][b]Modulos por defecto del Ultimate Portal[/b][/size][/color][/center]
o Aportes destacados (Greats Posts Module, Modulo especializado y desarrollado enteramente por vicram10, especial para los Foros del tipo Link Sharing)
o Noticias (NEWS)
o Noticias del Foro (Boards News)
o Descargas (Download Module)
o Paginas Internas (Internal Page Module)
o Afiliados (Affiliates Module)
o Quienes Somos (About Us Module)
o Preguntas Frecuentes (FAQs Module)

 * OBS: Los bloques y modulos puede aumentar, no seran solo esto, dependera de lo que se vaya agregando en el transcurso del desarrollo del portal.[center][color=orange][size=12pt][b]Administrador de Bloques[/b][/size][/color][/center]Ultimate Portal, cuenta con un Area de Administracion de bloques, facil, he intuitivo en su forma de uso. Dentro de esta area del panel de Administracion del Ultimate Portal puedes hacer lo siguiente:
 [list]
	[li][u]Posicion de los Bloques[/u]: de esta forma tendras una alta facilidad de configuracion del orden y activacion de cualquier bloque que se quiera usar dentro de Ultimate Portal ([url=http://www.ultimate-portal.net/image/blocks/admin-block-position.jpg][b]Ver Grafica[/b][/url])[/li]
	[li][u]Titulos de los Bloques[/u]: en esta zona, tendras la facilidad de establecer titulos diferentes para cada bloque ([url=http://www.smfsimple.com/img/ultimateportal/admin-block-title.jpg][b]Ver Grafica[/b][/url])[/li]
	[li][u]Crear Bloque[/u]: zona en la que podras crear tus propios bloques personalizados ([url=http://www.smfsimple.com/img/ultimateportal/admin-block-create.jpg][b]Ver Grafica[/b][/url]), tanto en forma de HTML ([url=http://www.smfsimple.com/img/ultimateportal/admin-block-html.jpg][b]Ver Grafica[/b][/url]) o creando bloques PHP ([url=http://www.smfsimple.com/img/ultimateportal/admin-block-php.jpg][b]Ver Grafica[/b][/url])[/li]
	[li][u]Administrando Bloques[/u]: en esta area ([url=http://www.smfsimple.com/img/ultimateportal/admin-block.jpg][b]Ver Grafica[/b][/url]), tendras entera disposicion de los bloques tanto del Sistema (Bloques por defecto), asi como los bloques personalizados (Bloques hechos por ustedes), podran: Asignarle permisos[/li]
[/list][list]
	[li]Asignarle Permisos: de esta forma, tendras entera facilidad de permitir que grupos de usuarios tendran derecho de visualizar el bloque creado.[/li]
	[li]Editar Bloques[/li]
	[li]Borrar Bloques[/li]
[/list] 

Todos los bloques que sean agregados en la Zona de Bloques, tendra su autoinstalador, simplemente usando el [b]Instalador de Paquetes de SMF ([/b]regla obligatoria[b]).[/b]

[hr]
[center][color=green][size=12pt][b]Por todo esto y mucho mas que no podemos explicar porque se haria tan largo el tema que seria mejor descargar el portal y probarlo! Ya no les quedarian dudas de que Ultimate Portal (UP para los amigos)  es el portal que todos estaban buscando![/b][/size][/color][/center]
[b][url=http://www.smfsimple.com]SMFSimple[/url][/b] agradece a todos los que ayudaron a hacer [b] Ultimate Portal[/b] lo que es hoy. 

[b]Fundador de Ultimate Portal y Director del proyecto:[/b] Victor "vicram10" Ramirez 

[b]Staff:[/b] vicram10, Lean, 4kstore & Distante. 

[b]Gracias especiales a[/b] Nino_16, royalduke, Suki, Liam, Near, Frony & Maliante! 

 [b]Gracias especiales por los iconos utilizados de: [/b] [url=http://www.famfamfam.com/lab/icons/silk/]FamFamFam[/url] | [url=http://dryicons.com/]DryIcons[/url] | [url=http://www.iconfinder.com/]Iconfinder[/url] 

 Y para todos aquellos que olvidamos saludar y que ayudaron al desarrollo del portal, [b]Muchas gracias[/b]

changelog.txt
Changelog:

CORREGIDO!!!
-Afiliados:
Corregidas las consultas, eliminados los globals que no se usan.

-bk-last-user.php
eliminadas algunas lineas. Agregado el array en la consulta, e indentado el codigo

-News
bknews.php -> Corregido un poco el codigo agregando algunos ifs para la secciones y eliminando cargas innecesarias.


-FaQ:
Permisos
Guardar los valores del settings
LoadFAQMain() - Sacadas consultas dentro del while...
UPFaq.php - Eliminados Globals no usados y estructuradas las consultas.

-Sub-UltimatePortal.php
Terminado: Borrar globales y $db_prefix que no se usan.

-Main Links:
LoadMainLinks() - Agregado un parametro para saber quien lo llama y en base a eso hacer la consulta mas liviana.
bk-menu.php - Agregado el parametro 

-Board News:
bk-boards-news.php - Sacados el loadmember para optimizarlo y globales.

-NEWS:
Globales y consultas..

-About:
-Globales

Revisar:

Por hacer!!!!
-Afiliados:
Ver el tema del movimiento ya que <marquee> no fue depreciado..

-UPNews:
ShowNewsMain() -> Intentar de unificar todo en una sola consulta y sacarlo de adentro del while.
DeleteNew() -> Revisar la redireccion para ver de optimizar.

-Sub-UltimatePortal.php
Revisar Todas las consultas
LoadID_MEMBER($memberName) -> Ponerle Limit 1 
upSSI_BoardNews() - Genera una consulta por cada tema que muestra... habria que optimizarlo..
LoadBlockNews() - Genera una consulta por cada noticia que muestra... habria que optimizarlo..
LoadStaffMembers() - Tiene consultas dentro de whiles y se podria usar el loadmemberdata.

UltimatePortal.php
+ Nuevo codigo de google analytic


install_db.php
<?php
/*---------------------------------------------------------------------------------
*	Ultimate Portal
*	Version 0.3
*	Project manager: vicram10
*	Copyright 2011
*	Powered by SMFSimple.com
**********************************************************************************/
	global $mbname, $boardurl, $db_prefix, $context;
	global $smcFunc, $db_name;
	// Define the Manual Installation Status
    $manual_install = false;
    if (file_exists(dirname(__FILE__) . '/SSI.php') && !defined('SMF')){
		require_once(dirname(__FILE__) . '/SSI.php');
	
		$manual_install = true;
    }
    elseif (!defined('SMF'))
	die('The Ultimate Portal installer wasn\'t able to connect to SMF! Make sure that you are either installing this via the Package Manager or the SSI.php file is in the same directory.');
    if ($manual_install)
	echo '<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml"><head>
	<title>Ultimate Portal Database Installer</title>
     <link rel="stylesheet" type="text/css" href="Themes/default/style.css" />
</head>
<body>
	<br /><br />';
	//Call db_extend
	db_extend('packages');
    // The Ultimate Portal Creating tables
	$tables = array(
		// UP Settings
		'ultimate_portal_settings' => array(
			'name' => 'ultimate_portal_settings',
			//Columns
			'columns' => array(
				array(
					'name' => 'variable',
					'type' => 'text',
				),							   
				array(
					'name' => 'value',
					'type' => 'text',
				),							   				
				array(
					'name' => 'section',
					'type' => 'text',
				),							   								
			),
			//End Columns
			//Primary?
			'indexes' => array(
				array(
					'type' => 'primary',
					'columns' => array('variable(30)')
				),
			)
			//End Primary
		),		
		//End Table
		// UP Main Link
		'ultimate_portal_main_links' => array(
			'name' => 'ultimate_portal_main_links',
			//Columns
			'columns' => array(
				array(
					'name' => 'id',
					'type' => 'int',
					'auto' => true,
					'unsigned' => true,
				),							   
				array(
					'name' => 'icon',
					'type' => 'text',
				),							   
				array(
					'name' => 'title',
					'type' => 'varchar',
					'size' => 255,
					'null' => 'not null',
					'default' => '',
				),				
				array(
					'name' => 'url',
					'type' => 'text',
				),							   				
				array(
					'name' => 'position',
					'type' => 'int',
					'size' => 2,
					'null' => 'not null',
					'unsigned' => true,
					'default' => 0,
				),							   								
				array(
					'name' => 'active',
					'type' => 'tinyint',
					'size' => 1,
					'null' => 'not null',
					'unsigned' => true,
					'default' => 0,
				),							   												
			),
			//End Columns
			//Primary?
			'indexes' => array(
				array(
					'type' => 'primary',
					'columns' => array('id')
				),
			)
			//End Primary
		),		
		//End Table		
		// UP Blocks
		'ultimate_portal_blocks' => array(
			'name' => 'ultimate_portal_blocks',
			//Columns
			'columns' => array(
				array(
					'name' => 'id',
					'type' => 'int',
					'size' => 11,
					'auto' => true,
					'unsigned' => true,
				),							   
				array(
					'name' => 'file',
					'type' => 'varchar',
					'size' => 255,
					'null' => 'null',
					'default' => '',
				),				
				array(
					'name' => 'title',
					'type' => 'varchar',
					'size' => 255,
					'null' => 'null',
					'default' => '',
				),				
				array(
					'name' => 'icon',
					'type' => 'varchar',
					'size' => 255,
					'null' => 'null',
					'default' => '',
				),					
				array(
					'name' => 'position',
					'type' => 'varchar',
					'size' => 20,
					'null' => 'not null',
					'default' => 'left',
				),								
				array(
					'name' => 'progressive',
					'type' => 'int',
					'size' => 3,
					'null' => 'not null',
					'unsigned' => true,
					'default' => 100,
				),					
				array(
					'name' => 'active',
					'type' => 'varchar',
					'size' => 10,
					'null' => 'null',
					'default' => '',
				),												
				array(
					'name' => 'personal',
					'type' => 'int',
					'size' => 2,
					'null' => 'not null',
					'unsigned' => true,
					'default' => 0,
				),							   												
				array(
					'name' => 'content',
					'type' => 'text',
				),				
				array(
					'name' => 'perms',
					'type' => 'text',
				),				
				array(
					'name' => 'bk_collapse',
					'type' => 'char',
					'size' => 5,
					'null' => 'null',
					'default' => '',
				),																
				array(
					'name' => 'bk_no_title',
					'type' => 'char',
					'size' => 5,
					'null' => 'null',
					'default' => '',
				),					
				array(
					'name' => 'bk_style',
					'type' => 'char',
					'size' => 5,
					'null' => 'null',
					'default' => '',
				),
				array(
					'name' => 'mbk_view',
					'type' => 'text',
				),				
			),
			//End Columns
			//Primary?
			'indexes' => array(
				array(
					'type' => 'primary',
					'columns' => array('id')
				),
			)
			//End Primary
		),		
		//End Table
		// UP Extra Fields 
		'uposts_extra_field' => array(
			'name' => 'uposts_extra_field',
			//Columns
			'columns' => array(
				array(
					'name' => 'id',
					'type' => 'int',
					'size' => 10,
					'auto' => true,
					'unsigned' => true,
				),							   
				array(
					'name' => 'title',
					'type' => 'varchar',
					'size' => 40,
					'null' => 'not null',
				),				
				array(
					'name' => 'icon',
					'type' => 'varchar',
					'size' => 255,
					'null' => 'not null',
				),				
				array(
					'name' => 'field',
					'type' => 'varchar',
					'size' => 100,
					'null' => 'not null',
					'default' => '0',
				),					
			),
			//End Columns
			//Primary?
			'indexes' => array(
				array(
					'type' => 'primary',
					'columns' => array('id')
				),
			)
			//End Primary
		),		
		//End Table					
		// UP Groups Perms
		'up_groups_perms' => array(
			'name' => 'up_groups_perms',
			//Columns
			'columns' => array(
				array(
					'name' => 'ID_GROUP',
					'type' => 'smallint',
					'size' => 5,
					'null' => 'not null',
					'default' => 0,
					'unsigned' => true,
				),							   
				array(
					'name' => 'permission',
					'type' => 'varchar',
					'size' => 30,
					'default' => '',
				),				
				array(
					'name' => 'value',
					'type' => 'tinyint',
					'size' => 4,
					'null' => 'not null',
					'default' => 1,
					'unsigned' => true,
				),							   				
			),
			//End Columns
			//Primary?
			'indexes' => array(
				array(
					'type' => 'primary',
					'columns' => array('ID_GROUP','permission')
				),
			)
			//End Primary
		),		
		//End Table						
		// UP News
		'up_news' => array(
			'name' => 'up_news',
			//Columns
			'columns' => array(
				array(
					'name' => 'id',
					'type' => 'int',
					'size' => 11,
					'auto' => true,
					'unsigned' => true,
				),				
				array(
					'name' => 'id_category',
					'type' => 'int',
					'size' => 10,
					'null' => 'not null',
					'default' => 0,
					'unsigned' => true,
				),				
				array(
					'name' => 'id_member',
					'type' => 'int',
					'size' => 10,
					'null' => 'not null',
					'default' => 0,
					'unsigned' => true,
				),						
				array(
					'name' => 'title',
					'type' => 'varchar',
					'size' => 255,
					'null' => 'not null',
				),				
				array(
					'name' => 'username',
					'type' => 'varchar',
					'size' => 34,
					'null' => 'not null',
				),				
				array(
					'name' => 'body',
					'type' => 'text',
				),				
				array(
					'name' => 'date',
					'type' => 'int',
					'size' => 10,
					'null' => 'not null',
					'default' => 0,
					'unsigned' => true,
				),					
				array(
					'name' => 'id_member_updated',
					'type' => 'int',
					'size' => 10,
					'null' => 'null',
					'unsigned' => true,
				),						
				array(
					'name' => 'username_updated',
					'type' => 'varchar',
					'size' => 34,
					'null' => 'null',
				),								
				array(
					'name' => 'date_updated',
					'type' => 'int',
					'size' => 10,
					'null' => 'null',
					'unsigned' => true,
				),										
			),
			//End Columns
			//Primary?
			'indexes' => array(
				array(
					'type' => 'primary',
					'columns' => array('id')
				),
			)
			//End Primary
		),		
		//End Table
		// UP News Section
		'up_news_sections' => array(
			'name' => 'up_news_sections',
			//Columns
			'columns' => array(
				array(
					'name' => 'id',
					'type' => 'int',
					'size' => 10,
					'auto' => true,
					'unsigned' => true,
				),											   
				array(
					'name' => 'title',
					'type' => 'varchar',
					'size' => 40,
					'null' => 'not null',
				),												
				array(
					'name' => 'icon',
					'type' => 'varchar',
					'size' => 255,
					'null' => 'not null',
				),																
				array(
					'name' => 'position',
					'type' => 'int',
					'size' => 4,
					'null' => 'not null',
					'default' => 0,
					'unsigned' => true,
				),														
			),
			//End Columns
			//Primary?
			'indexes' => array(
				array(
					'type' => 'primary',
					'columns' => array('id')
				),
			)
			//End Primary
		),		
		//End Table		
	
		
		//End Table								
		// UP Internal Page
		'ultimate_portal_ipage' => array(
			'name' => 'ultimate_portal_ipage',
			//Columns
			'columns' => array(
				array(
					'name' => 'id',
					'type' => 'int',
					'size' => 11,
					'auto' => true,
					'unsigned' => true,
				),		
				array(
					'name' => 'title',
					'type' => 'varchar',
					'size' => 255,
					'default' => '',
				),				
				array(
					'name' => 'sticky',
					'type' => 'int',
					'size' => 3,
					'default' => 0,
					'unsigned' => true,
				),		
				array(
					'name' => 'active',
					'type' => 'varchar',
					'size' => 10,
					'default' => '',
				),								
				array(
					'name' => 'type_ipage',
					'type' => 'text',
				),												
				array(
					'name' => 'content',
					'type' => 'text',
				),							
				array(
					'name' => 'perms',
					'type' => 'text',
				),							
				array(
					'name' => 'column_left',
					'type' => 'int',
					'size' => 1,
					'default' => 0,
					'unsigned' => true,
				),														
				array(
					'name' => 'column_right',
					'type' => 'int',
					'size' => 1,
					'default' => 0,
					'unsigned' => true,
				),														
				array(
					'name' => 'date_created',
					'type' => 'int',
					'size' => 10,
					'null' => 'null',
					'unsigned' => true,
				),																		
				array(
					'name' => 'date_updated',
					'type' => 'int',
					'size' => 10,
					'null' => 'null',
					'unsigned' => true,
				),	
				array(
					'name' => 'id_member',
					'type' => 'int',
					'size' => 10,
					'null' => 'not null',
					'unsigned' => true,
				),		
				array(
					'name' => 'username',
					'type' => 'tinytext',
				),											
				array(
					'name' => 'id_member_updated',
					'type' => 'int',
					'size' => 10,
					'null' => 'null',
					'unsigned' => true,
				),		
				array(
					'name' => 'username_updated',
					'type' => 'tinytext',
				),											
			),
			//End Columns
			//Primary?
			'indexes' => array(
				array(
					'type' => 'primary',
					'columns' => array('id')
				),
			)
			//End Primary
		),		
		//End Table	
						
			
		// UP Multiblock
		'up_multiblock' => array(
			'name' => 'up_multiblock',
			//Columns
			'columns' => array(
				array(
					'name' => 'id',
					'type' => 'int',
					'size' => 11,
					'auto' => true,
					'unsigned' => true,
				),		
				array(
					'name' => 'title',
					'type' => 'text',
				),											
				array(
					'name' => 'blocks',
					'type' => 'text',
				),											
				array(
					'name' => 'position',
					'type' => 'text',
				),											
				array(
					'name' => 'design',
					'type' => 'text',
				),
				array(
					'name' => 'mbk_title',
					'type' => 'char',
					'size' => 5,
					'null' => 'null',
					'default' => '',
				),				
				array(
					'name' => 'mbk_collapse',
					'type' => 'char',
					'size' => 5,
					'null' => 'null',
					'default' => '',
				),							
				array(
					'name' => 'mbk_style',
					'type' => 'char',
					'size' => 5,
					'null' => 'null',
					'default' => '',
				),		
				array(
					'name' => 'order',
					'type' => 'int',
					'size' => 3,
					'null' => 'null',
					'unsigned' => true,
				),				
				array(
					'name' => 'enable',
					'type' => 'smallint',
					'size' => 1,
					'null' => 'null',
					'unsigned' => true,
				),				
			),
			//End Columns
			//Primary?
			'indexes' => array(
				array(
					'type' => 'primary',
					'columns' => array('id')
				),
			)
			//End Primary
		),
		//End Table
	);	

	//Creating Tables
	foreach ($tables as $table)
	{
		$table_name = $table['name'];
		$smcFunc['db_create_table']('{db_prefix}' . $table_name, $table['columns'], $table['indexes']);		
		$currentTable = $smcFunc['db_table_structure']('{db_prefix}' . $table_name);
		// Check that all columns are in
		foreach ($table['columns'] as $id => $col)
		{
			$exists = false;
			// TODO: Check that definition is correct
			foreach ($currentTable['columns'] as $col2)
			{
				if ($col['name'] === $col2['name'])
				{
					$exists = true;
					break;
				}
			}

			// Add missing columns
			if (!$exists)
				$smcFunc['db_add_column']('{db_prefix}' . $table_name, $col);

			//Check, not change anything?
			if($exists)
			{
				$smcFunc['db_change_column']('{db_prefix}' . $table_name, $col['name'], $col);
			}

		}
		//End add missing columns
		// Check that all indexes are in and correct
		foreach ($table['indexes'] as $id => $index)
		{
			$exists = false;

			foreach ($currentTable['indexes'] as $index2)
			{
				// Primary is special case
				if ($index['type'] == 'primary' && $index2['type'] == 'primary')
				{
					$exists = true;

					if ($index['columns'] !== $index2['columns'])
					{
						$smcFunc['db_remove_index']('{db_prefix}' . $table_name, 'primary');
						$smcFunc['db_add_index']('{db_prefix}' . $table_name, $index);
					}

					break;
				}
				// Make sure index is correct
				elseif (isset($index['name']) && isset($index2['name']) && $index['name'] == $index2['name'])
				{
					$exists = true;

					// Need to be changed?
					if ($index['type'] != $index2['type'] || $index['columns'] !== $index2['columns'])
					{
						$smcFunc['db_remove_index']('{db_prefix}' . $table_name, $index['name']);
						$smcFunc['db_add_index']('{db_prefix}' . $table_name, $index);
					}

					break;
				}
			}

			if (!$exists)
				$smcFunc['db_add_index']('{db_prefix}' . $table_name, $index);
		}
		//End check indexes
	}

	//Install default rows
	$smcFunc['db_insert']('replace',
		'{db_prefix}ultimate_portal_settings',
		array('variable' => 'text', 'value' => 'text', 'section' => 'text'),
		array(
			array('ultimate_portal_enable', 'on', 'config_preferences'),
			array('ultimate_portal_home_title', ''. $mbname .' - Home', 'config_preferences'),
			array('favicons', '', 'config_preferences'),
			array('up_use_curve_variation', 'on', 'config_preferences'),
			array('ultimate_portal_width_col_left', '20%', 'config_preferences'),
			array('ultimate_portal_width_col_center', '60%', 'config_preferences'),
			array('ultimate_portal_width_col_right', '20%', 'config_preferences'),
			array('ultimate_portal_enable_icons', 'on', 'config_preferences'),
			array('ultimate_portal_enable_version', 'on', 'config_preferences'),
			array('ultimate_portal_enable_col_left', 'on', 'config_preferences'),
			array('ultimate_portal_enable_col_right', 'on', 'config_preferences'),
			array('up_forum_enable_col_left', 'on', 'config_preferences'),
			array('up_forum_enable_col_right', 'on', 'config_preferences'),
			array('ultimate_portal_icons_extention', '.png', 'config_preferences'),
			array('board_news_limit', '5', 'config_board_news'),
			array('board_news_view', '0', 'config_board_news'),
			array('board_news_lenght', '2000', 'config_board_news'),
			array('up_news_enable', 'on', 'config_up_news'),
			array('up_news_limit', '5', 'config_up_news'),						
			array('up_news_global_announcement', '', 'config_up_news'),
			array('ipage_enable', '', 'config_ipage'),
			array('ipage_active_columns', '', 'config_ipage'),
			array('ipage_social_bookmarks', 'on', 'config_ipage'),
			array('ipage_limit', '10', 'config_ipage'),			
			array('seo_google_analytics', '', 'config_seo'),
			array('seo_title_keyword', '', 'config_seo'),
			array('seo_google_verification_code', '', 'config_seo'),			
			array('up_reduce_site_overload', '', 'config_preferences'),
			array('up_left_right_collapse', 'on', 'config_preferences'),
		),
		array('variable')
	);
	//End Add rows in UP settings table
	
	//smf_ultimate_portal_main_links INSERT
	$smcFunc['db_insert']('ignore',
		'{db_prefix}ultimate_portal_main_links',
		array('id' => 'int', 'icon' => 'text', 'title' => 'string',  'url' => 'text', 'position' => 'int', 'active' => 'int'),
		array(
			array(1, '<UP_MAIN_LINK_ICON>/home.png', 'Home', '<UP_BOARDURL>/index.php', 1, 1),
			array(2, '<UP_MAIN_LINK_ICON>/forum.png', 'Forum', '<UP_BOARDURL>/index.php?action=forum', 2, 1),
			array(3, '<UP_MAIN_LINK_ICON>/news.png', 'News', '<UP_BOARDURL>/index.php?action=news', 3, 0),			
			array(4, '<UP_MAIN_LINK_ICON>/internal-page.png', 'Internal Page', '<UP_BOARDURL>/index.php?action=internal-page', 6, 0),			
		),
		array('id')
	);
	//End insert rows in Main links Table
	
	//Insert rows in Blocks table
	$smcFunc['db_insert']('ignore',
		'{db_prefix}ultimate_portal_blocks',
		array('id' => 'int', 'file' => 'string', 'title' => 'string',  'icon' => 'string', 'position' => 'string', 'progressive' => 'int', 'active' => 'string', 
			  'personal' => 'int', 'content' => 'text', 'perms' => 'text', 'bk_collapse' => 'string', 'bk_no_title' => 'string', 'bk_style' => 'string', 'mbk_view' => 'text'),
		array(
			array(1, 'bk-boards-news.php', 'Board News', 'bk-boards-news', 'center', 2, 'checked', 0, '', '', 'on', '', 'on', ''),
			array(2, 'bk-personal-menu.php', 'User', 'bk-personal-menu', 'left', 2, 'checked', 0, '', '', 'on', '', 'on', ''),
			array(3, 'bk-menu.php', 'Menu Portal', 'bk-menu', 'left', 1, 'checked', 0, '', '', 'on', '', 'on', ''),
			array(4, 'bk-news.php', 'News', 'bk-news', 'center', 1, '', 0, '', '', 'on', '', 'on', ''),
			array(5, 'bk-online.php', 'Users Online', 'bk-online', 'left', 3, 'checked', 0, '', '', 'on', '', 'on', ''),
			array(6, 'bk-site-stats.php', 'Stats', 'bk-site-stats', 'right', 2, 'checked', 0, '', '', 'on', '', 'on', ''),
			array(7, 'bk-top-poster.php', 'Top Posters', 'bk-top-poster', 'right', 1, 'checked', 0, '', '', 'on', '', 'on', ''),
			array(8, 'bk-lastUsers.php', 'Last Users', 'bk-lastUsers', 'right', 4, 'checked', 0, '', '', 'on', '', 'on', ''),
			array(9, 'bk-recent-topics.php', 'Recent Topics', 'bk-recent-topics', 'right', 3, 'checked', 0, '', '', 'on', '', 'on', ''),
		),
		array('id')
	);	
	//End insert rows in Blocks table
	
	// OK, time to report, output all the stuff to be shown to the user
	if ($manual_install){
echo '
<table cellpadding="0" cellspacing="0" border="0" class="tborder" width="800" align="center"><tr><td>
<div class="titlebg" style="padding: 1ex" align="center">
	Ultimate Portal Database Installer
</div>
<div class="windowbg2" style="padding: 2ex">
<div style="padding-top:30px">
	<strong>Your database update has been completed successfully!</strong>
	<br /><br />The Ultimate Portal Database was successfully installed.<br />
	Now you should go to the <a href="', $scripturl, '?action=admin;area=preferences">Ultimate Portal</a>.
	<br /><br />
	<span style="color: #FF0000"><strong>Please NOW Delete this file. </strong></span>
</div>
</td></tr></table>
<br />
</body></html>';
    }
?>